tedana/viz.py

```diff
@@ -32,55 +60,60 @@ def write_comp_figs(ts, mask, comptable, mmix, n_vols,
     mmix : (C x T) array_like
         Mixing matrix for converting input data to component space, where `C`
         is components and `T` is the same as in `data`
-    n_vols : :obj:`int`
-        Number of volumes in original time series
-    acc : :obj:`list`
-        Indices of accepted (BOLD) components in `mmix`
-    rej : :obj:`list`
-        Indices of rejected (non-BOLD) components in `mmix`
-    midk : :obj:`list`
-        Indices of mid-K (questionable) components in `mmix`
-    empty : :obj:`list`
-        Indices of ignored components in `mmix`
     ref_img : :obj:`str` or img_like
         Reference image to dictate how outputs are saved to disk
 
     """
+    # Get the lenght of the timeseries
+    n_vols = len(mmix)
 
+    # Check that colormap provided exists
+    if png_cmap not in plt.colormaps():
+        LGR.warning('Provided colormap is not valid, proceeding with coolwarm')
+        png_cmap = 'coolwarm'
     # regenerate the beta images
     ts_B = model.get_coeffs(ts, mmix, mask)
     ts_B = ts_B.reshape(ref_img.shape[:3] + ts_B.shape[1:])
-    # Mask out zeros
+    # trim edges from ts_B array
+    ts_B = trim_edge_zeros(ts_B)
+
+    # Mask out remaining zeros
     ts_B = np.ma.masked_where(ts_B == 0, ts_B)
 
     # Get repetition time from ref_img
     tr = ref_img.header.get_zooms()[-1]
 
-    # Start making plots
-    if not os.path.isdir('figures'):
-        os.mkdir('figures')
-
     # Create indices for 6 cuts, based on dimensions
     cuts = [ts_B.shape[dim] // 6 for dim in range(3)]
+    expl_text = ''
 
+    # Remove trailing ';' from rationale column
+    comptable['rationale'] = comptable['rationale'].str.rstrip(';')
     for compnum in range(0, mmix.shape[1], 1):
 
-        if compnum in acc:
+        if comptable.iloc[compnum]["classification"] == 'accepted':
             line_color = 'g'
-        elif compnum in rej:
+            expl_text = 'accepted'
+        elif comptable.iloc[compnum]["classification"] == 'rejected':
             line_color = 'r'
-        elif compnum in midk:
-            line_color = 'm'
-        else:
+            expl_text = 'rejection reason(s): ' + comptable.iloc[compnum]["rationale"]
+        elif comptable.iloc[compnum]["classification"] == 'ignored':
             line_color = 'k'
+            expl_text = 'ignored reason(s): ' + comptable.iloc[compnum]["rationale"]
+        else:
+            # Classification not added
+            # If new, this will keep code running
+            line_color = '0.75'
+            expl_text = 'unknown classification'
```

Comment:
Instead of "Unknown" could we say "Other" ? I ask since we likely will _know_ what new classifications are, even if they don't fit into these bins.

Reply:
I'm good with other